internal/processor/filterspan/filterspan.go

```diff
@@ -40,17 +44,28 @@ type propertiesMatcher struct {
 	// Span names to compare to.
 	nameFilters filterset.FilterSet
 
+	// Instrumentation libraries to compare against
+	Libraries []instrumentationLibraryMatcher
+
 	// The attribute values are stored in the internal format.
 	Attributes attributesMatcher
+
+	Resources attributesMatcher
+}
+
+type instrumentationLibraryMatcher struct {
+	Name    filterset.FilterSet
+	Version filterset.FilterSet
 }
 
 type attributesMatcher []attributeMatcher
 
 // attributeMatcher is a attribute key/value pair to match to.
 type attributeMatcher struct {
 	Key string
-	// If nil only check for key existence.
+	// If both AttributeValue and StringFilter are nil only check for key existence.
 	AttributeValue *pdata.AttributeValue
+	StringFilter   filterset.FilterSet
```

Comment:
Unclear what this field does. We already have a Key and an AttributeValue which supposedly is all we can match in an attribute. Add a comment to explain.